### DISCUSSION TITLE
NGINX Proxy with local access behind the scenes

### DISCUSSION THREAD

**Discussion by terrypearson** [openobserve/openobserve#5235] [Q&A]
Hello,

I have a domain name: example.com
I want to put openobserve at example.com/somestuff/open but I still want to be able to push log entries directly from my webservers through the data center's local network instead of using the outside interface.

So using filebeat, I had setup something like this:
`
output.elasticsearch:
  hosts: ["http://10.11.11.11:5080/utils/somestuff/open/"]
  timeout: 10
  path: "/api/default/"
  index: "default"
`
In openobserve, I set the ZO_BASE_URI.

This worked quite well, until I wanted to put OpenObserve behind a file path on my webserver. I found this: https://openobserve.ai/docs/operator-guide/nginx_proxy/ which was helpful in getting the front end working on my nginx proxy. But once I did that, I began getting these errors from my servers that were trying to connect to OpenObserve:
`"Attempting to reconnect to backoff(elasticsearch(http://10.11.11.11:5080/utils/nothere/openobserve/)) with 15 reconnect attempt(s)"`

Basically, it can't connect to the local interface after you set ZO_BASE_DIR. 

**Question**
I guess my question is if there is a way to get openobserve to use the ZO_BASE_URI for front end stuff, but to still be able to send logs through the local network interface directly to the server?

**Comment by hengfeiyang** [comment]
try this:
```
output.elasticsearch: 
    hosts: ["http://10.11.11.11:5080"] 
    path: "/ZO_BASE_URI/api/default/" 
    index: "default"
    timeout: 10 
```

**Reply by terrypearson** [reply]
Thank you for your help. However, that still left me with a 404 error when attempting to push data using filebeat. I actually tried both of these options in path:
`path: "/ZO_BASE_URI/api/default/"`
`path: "http://10.11.11.11:5080/api/default/"`

I don't see a way where it is possible to use openobserve behind a proxy but then use a local ip address at your data center to send data.  However, my solution was to use a subdomain instead of a file path. This solved it for me.  I would encourage checking how openobserve works behind a proxy, especially when the logs are being sent over the local interface.

Thank you again for your help. While it did not exactly work, I was able to solve it with a subdomain. Have a nice day.

**Reply by hengfeiyang** [reply]
What is the 404 error? what is the log like? and what is your ZO_BASE_URI?
